Assets/Scripts/HotBar/SlotSwitch.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class SlotSwitch : MonoBehaviour
+{
+    // Start is called before the first frame update
+
+    private List<GameObject> slotsBg;
+    private int current = 0;
+    void Start()
+    {
+        slotsBg = new List<GameObject>();
+        for (int i = 0; i < 4; i++)
+        {
+            slotsBg.Add(transform.GetChild(i).gameObject);
+        }
+        Image img = slotsBg[0].GetComponent<Image>();
+        Color col = img.color;
+        col.a = 0.5f;
+        img.color = col;
+        current = 0;
+    }
+
+    // Update is called once per frame
+    void Update()
```

Comment:
please remove duplicated code for the slots : only Image img = slotsBg[ ].GetComponent<Image>(); differs